ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/deduplication/DeduplicationPeriodService.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.deduplication
+
+import java.time.{Duration, Instant}
+
+import akka.stream.Materializer
+import com.daml.error.ContextualizedErrorLogger
+import com.daml.ledger.api.DeduplicationPeriod
+import com.daml.ledger.api.domain.ApplicationId
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+import com.daml.platform.server.api.validation.{DeduplicationPeriodValidator, ErrorFactories}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeduplicationPeriodService(
+    converter: DeduplicationPeriodConverter,
+    validation: DeduplicationPeriodValidator,
+    errorFactories: ErrorFactories,
+) {
+
+  def supportedDeduplicationPeriod(
+      deduplicationPeriod: DeduplicationPeriod,
+      maxDeduplicationDuration: Duration,
+      applicationId: ApplicationId,
+      actAs: Set[Ref.Party],
+      submittedAt: Instant,
+  )(implicit
+      mat: Materializer,
+      ec: ExecutionContext,
+      loggingContext: LoggingContext,
+      contextualizedErrorLogger: ContextualizedErrorLogger,
+  ): Future[DeduplicationPeriod] = {
+    val validatedDeduplicationPeriod = deduplicationPeriod match {
+      case period: DeduplicationPeriod.DeduplicationDuration =>
+        Future.successful(validation.validate(period, maxDeduplicationDuration))
```

Comment:
Why do we validate this one again? Just for cosistency? I assume we just have to validate the duration from an offset?

Reply:
That is correct, just to be consistent in the way we handle them

Reply:
This is not clear to me. This logic is part of the KV participant state implementation and it runs as part of the participant that has already validated the period. Why should we do it again?

Reply:
Removed it.